testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java

```diff
@@ -224,6 +224,35 @@ public void testUpdate()
         onTrino().executeQuery("DROP TABLE " + icebergTableName);
     }
 
+    @Test(groups = {HIVE_ICEBERG_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testMerge()
+    {
+        String sourceTableName = "iceberg_merge_source_" + randomTableSuffix();
+        String targetTableName = "iceberg_merge_target_" + randomTableSuffix();
+        String hiveSourceTableName = "hive.default." + sourceTableName;
+        String hiveTargetTableName = "hive.default." + targetTableName;
+        String icebergSourceTableName = "iceberg.default." + sourceTableName;
+        String icebergTargetTableName = "iceberg.default." + targetTableName;
+
+        createIcebergTable(icebergSourceTableName, true, true);
+        createIcebergTable(icebergTargetTableName, true, false);
+
+        assertThat(onTrino().executeQuery("""
+                    MERGE INTO %s t USING %s s
+                    ON t.nationkey = s.nationkey
+                    WHEN NOT MATCHED
+                        THEN INSERT (nationkey, name, regionkey, comment)
+                                VALUES (s.nationkey, s.name, s.regionkey, s.comment)
+                """.formatted(hiveTargetTableName, hiveSourceTableName)
+        )).updatedRowsCountIsEqualTo(25);
```

Comment:
checkstyle should error here

Reply:
i see it didn't, but we have a rule that disallows whitespace before `)`, right?

Reply:
I think you are hinting at https://github.com/airlift/airbase/pull/323
I can convert the text block to a string concat construct.

Reply:
> I think you are hinting at https://github.com/airlift/airbase/pull/323

no

> I can convert the text block to a string concat construct.

no need to.

the `))` should be on the previous line though